modules/tftp/server.rb

```diff
@@ -95,16 +107,88 @@ def pxeconfig_file(mac)
   end
 
   class Pxegrub2 < Server
-    def pxeconfig_dir
-      "#{path}/grub2"
+    def bootloader_path(os, version, arch)
+      unless (bootloader_universe = Proxy::TFTP::Plugin.settings.bootloader_universe)
+        logger.debug "TFTP: bootloader universe not configured."
+        return
+      end
+
+      bootloader_path = "#{bootloader_universe}/pxegrub2/#{os}/#{version}/#{arch}"
+
+      logger.debug "TFTP: Checking bootloader universe for suitable bootloader directory for"
+      logger.debug "TFTP:   * Operating system: #{os}"
+      logger.debug "TFTP:   * Version: #{version}"
+      logger.debug "TFTP:   * Architecture: #{arch}"
+      logger.debug "TFTP: Checking bootloader universe if \"#{bootloader_path}\" exists."
+      unless Dir.exist?(bootloader_path)
+        logger.debug "TFTP: Directory \"#{bootloader_path}\" does not exist."
+
+        bootloader_path = "#{bootloader_universe}/pxegrub2/#{os}/default/#{arch}"
+        logger.debug "TFTP: Checking if fallback directory at \"#{bootloader_path}\" exists."
+        unless Dir.exist?(bootloader_path)
+          logger.debug "TFTP: Directory \"#{bootloader_path}\" does not exist."
+          return
+        end
+      end
+
+      bootloader_path
+    end
+
+    def setup_bootloader(mac, os, major, minor, arch, bootfilename_efi, build)
+      pxeconfig_dir_mac = pxeconfig_dir(mac)
+
+      if build == "true"
+        logger.debug "TFTP: Host is in build mode."
+        logger.debug "TFTP:   => Deploying host specific bootloader files to \"#{pxeconfig_dir_mac}\""
+
+        FileUtils.mkdir_p(pxeconfig_dir_mac)
+
+        version = "#{major}#{".#{minor}" unless minor.empty?}"
```

Comment:
I think this is hard to read and would suggest
```suggestion
        version = (minor && !minor.empty?) ? "#{major}.#{minor}" : major
```